actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -42,16 +43,17 @@ static bool IsInWhatsNewDirectory(PullRequestFile file)
 // Also, filter out files that are part of the "What's new" directory - as they shouldn't require redirects.
 foreach (PullRequestFile file in files.Where(f => IsYmlOrMarkdownFile(f) && !IsInWhatsNewDirectory(f)))
 {
+    Matcher? matcher = DocfxConfigurationReader.GetMatcher();
     // Changing the extension from .yml to .md or the opposite doesn't require a redirection.
     // In both cases, the URL in live docs site is the same.
-    if (file.IsRenamed() && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
+    if (file.IsRenamed() && matcher?.Match(file.PreviousFileName)?.HasMatches == true && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
```

Comment:
Is the `== true` part in `matcher?.Match(file.PreviousFileName)?.HasMatches == true`, actually needed?

```suggestion
    if (file.IsRenamed() && matcher?.Match(file.PreviousFileName)?.HasMatches && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
```

Reply:
@IEvangelist Yes it's needed because the result of the left expression is nullable bool